relayer/message_relayer.go

```diff
@@ -548,27 +586,176 @@ func (r *messageRelayer) aggregateSignatures(signatureMap map[int]blsSignatureBu
 	return aggSig, vdrBitSet, nil
 }
 
+//
+// Database access
+//
+
+// Determines the height to process from. There are three cases:
+// 1) The database contains the latest processed block data for the chain
+//   - In this case, we return the maximum of the latest processed block and the configured processHistoricalBlocksFromHeight
```

Comment:
In what case would processHistoricalBlocksFromHeight > latest processed block?

Reply:
That would occur only if the user sets `processHistoricalBlocksFromHeight` to be greater than the stored height. If, for instance, the user intentionally takes their relayer offline for a period of time and doesn't want to process messages below a certain height.

The max function is much more relevant to the case in which a user sets `processHistoricalBlocksFromHeight` on initial startup, but we want to resume from the latest processed height rather than the configured height on subsequent startups. Taking the max of `processHistoricalBlocksFromHeight` and the stored height makes this possible without having to change the config.